mobile/src/main/java/ch/epfl/sdp/mobile/state/Navigation.kt

```diff
@@ -27,18 +35,29 @@ fun Navigation() {
 /** An interface representing the current state of the user. */
 private sealed interface UserState {
 
+  /**
+   * A unique key for this [UserState], which indicates that a full-screen transition should occur.
+   */
+  val key: Any
```

Comment:
[Question] Why is key of type `Any` ? Wouldn't be more concise if it was an `enum` instead, or simply a `String` ?

Reply:
The function `contentKey` from `AnimatedContent` returns an `Any`, so I thought I could simply match its return type. I don't think making it a different type would make the code more concise, but I could indeed enforce that it's a `String`.

However, I would not want future consumers of `UserState` to assume that they'll always get a `key` that's a `String`, so `Any` sounds like a good choice too. What are your thoughts on this ?

Reply:
What do you mean by future consumers ?

Reply:
Some code we write that uses `UserState`

Reply:
To me, the fact that a `key` of a `UserState` is a `String` is an implementation detail, so I don't think we should expose a more precise type

Reply:
Yes sounds logical